remotecommanddispatcher/src/main/java/com/tealium/remotecommanddispatcher/RemoteCommandConfigRetriever.kt

```diff
@@ -22,27 +25,39 @@ class AssetRemoteCommandConfigRetriever(
     private val loader: Loader = JsonLoader.getInstance(config.application)
 ) : RemoteCommandConfigRetriever {
 
-    private var _remoteCommandConfig: RemoteCommandConfig = loadConfig()
-    override val remoteCommandConfig: RemoteCommandConfig
+    private var _remoteCommandConfig: RemoteCommandConfig? = loadConfig()
+    override val remoteCommandConfig: RemoteCommandConfig?
         get() = _remoteCommandConfig
 
-    private fun loadConfig(): RemoteCommandConfig {
+    private fun loadConfig(): RemoteCommandConfig? {
         return loadFromAsset(filename)?.also {
             Logger.dev(BuildConfig.TAG, "Loaded local remote command settings.")
-        } ?: RemoteCommandConfig()
+        }
     }
 
     private fun loadFromAsset(filename: String): RemoteCommandConfig? {
-        return loader.loadFromAsset(filename)?.let {
-            try {
-                val json = JSONObject(it)
-                RemoteCommandConfig.fromJson(json)
-            } catch (ex: JSONException) {
-                Logger.qa(
-                    BuildConfig.TAG,
-                    "Error loading RemoteCommandsConfig JSON from asset: ${ex.message}"
-                )
-                null
+        return loadFromAsset(loader, filename)
+    }
+
+    override suspend fun refreshConfig() {
+        // nothing to do - asset is already loaded if valid.
+    }
+
+    companion object {
+        fun loadFromAsset(loader: Loader, filename: String): RemoteCommandConfig? {
+            val fullFilename = if (filename.endsWith(".json")) filename else "$filename.json"
```

Comment:
This is slightly different from iOS. Do you think it's worth unifying?
On [this branch](https://github.com/Tealium/tealium-kotlin/compare/main...mt-1504#diff-d1cb04d09ebe7d0a0bb6d6c81691614a71da7c69851db047767a464e9887895eR30-R36) I had tried to do the same change that I did on iOS which expands support for also files without extensions or files with a different extension (like .JSON in all caps)